source/installation/install_from_source.rst

```diff
@@ -12,10 +12,10 @@ running. For more elaborate information, please consult the rest of the
 How to make your life easier...
 -------------------------------
 
-There are :doc:`packages for rsyslog <packages>` available.
```

Comment:
same comment on hyperlinks with proper text. IMHO we should not sacrifice well-written html for the sake of replacement strings, if that's the issue. But again, I am open to discussion.

Reply:
Here I think I just moved the text. Instead of leading in with "there are packages available", I moved that below with the next section.

Also, a note on the link generation. I think the [sphinx manual](http://www.sphinx-doc.org/en/stable/markup/inline.html) says it better than I probably can:

> :doc:
> Link to the specified document; the document name can be specified in absolute or relative fashion. For example, if the reference :doc:`parrot` occurs in the document sketches/index, then the link refers to sketches/parrot. If the reference is :doc:`/people` or :doc:`../people`, the link refers to people.
> 
> If no explicit link text is given (like usual: :doc:`Monty Python members </people>`), **the link caption will be the title of the given document**.

The part I'm trying to take advantage of is the **the link caption will be the title of the given document** behavior. If the document I am linking too has a good title, I can give a bare reference to the document instead of explicit text that may mask a broken link.

Diff:

```diff
-:doc:`colorful birds that talk back<parrot>`
+:doc:`parrot`
```

In this case the page title is "Installing rsyslog from Package", so I thought a good way to link to the doc was to make use of the page title in the text that was going to link to it. We *could* override the page title and reference it earlier as was done before my edits, but I was attempting to make linking to the "Installing rsyslog from Package" more natural.

The substitution references and definitions is mostly an effort on my part to keep from repeating common phrases, especially if there is a desire to use them more than once. I've already seen in many places in the current version of the docs where content has been repeated and will need to be changed in multiple places. Investing in substitution definitions/references are part of an effort on my part to follow the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

It's a difficult balance, but that is my aim.

Reply:
ah, I didn't know it automatically pulls in the page title. So this probably addresses most of my concerns (assuming the title is OK).